share/doc/homebrew/Formula-Cookbook.md

```diff
@@ -588,24 +507,11 @@ class Foo < Formula
 end
 ```
 
-Formulae that only have `head` versions should be submitted to [homebrew/headonly](https://github.com/Homebrew/homebrew-headonly) instead of Homebrew/homebrew.
-
-### devel
-
-The "devel" spec (activated by passing `--devel`) is used for a project’s unstable releases. It is specified in a block:
-
-```ruby
-devel do
-  url "https://foo.com/foo-0.1.tar.gz"
-  sha256 "85cc828a96735bdafcf29eb6291ca91bac846579bcef7308536e0c875d6c81d7"
-end
-```
-
-You can test if the "devel" spec is in use with `build.devel?`.
+Formulae that only have [`head`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#head-class_method) versions should be submitted to [homebrew/headonly](https://github.com/Homebrew/homebrew-headonly) instead of Homebrew/homebrew.
 
 ## Compiler selection
 
-Sometimes a package fails to build when using a certain compiler. Since recent Xcode no longer includes a GCC compiler, we cannot simply force the use of GCC. Instead, the correct way to declare this is the `fails_with` DSL method. A properly constructed `fails_with` block documents the latest compiler build version known to cause compilation to fail, and the cause of the failure. For example:
+Sometimes a package fails to build when using a certain compiler. Since recent Xcodes no longer include a GCC compiler we cannot simply force the use of GCC. Instead, the correct way to declare this is the [`fails_with` DSL method](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#fails_with-class_method). A properly constructed [`fails_with`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#fails_with-class_method) block documents the latest compiler build version known to cause compilation to fail, and the cause of the failure. For example:
```

Comment:
> cause is a no-op

We may want to store such information in the future. Beside, I think it's a nice DSL which can encourage/remind authors to document the cause then normal comment(which feels more optional).